### PR TITLE
security: up dependencies to fix security vulnerabilities

### PR DESCRIPTION
**Pull request by sicoyle**
# Description

Dapr has a few vulnerabilities that we should address from a security perspective by upping our dependency versions 👇

| Package | Affected version | Patched version | Fixed in this PR? | Impact |
| -------- | ------- | ------- | ------- | ------- |
| google.golang.org/grpc | < 1.56.3 | v1.56.3 | yes | gRPC-Go HTTP/2 rapid reset vulnerability |
| [golang.org/x/net](http://golang.org/x/net) | < 0.17.0 | latest | yes | HTTP/2 rapid reset can cause excessive work in net/http | 
| github.com/hamba/avro/v2 | < 2.13.0 | latest | yes | avro vul to detail service via attacker-controller param | 
| [github.com/nats-io/nats-server/v2](http://github.com/nats-io/nats-server/v2) | >= 2.2.0, < 2.9.23 | latest | yes | NATS.io: Adding accounts for just the system account adds auth bypass |
| [github.com/jackc/pgx/v5](http://github.com/jackc/pgx/v5) | >= 5.0.0, < 5.5.4 | v5.5.4 | yes | pgproto3 SQL injection via protocol message size overflow | 
| [google.golang.org/protobuf](http://google.golang.org/protobuf) | < 1.33.0 | latest | Golang protojson.Unmarshal function infinite loop when unmarshaling certain forms of invalid JSON |
| [github.com/lestrrat-go/jwx/v2](http://github.com/lestrrat-go/jwx/v2) | < 2.0.21 | latest | JWX vulnerable to a denial of service attack using compressed JWE message  |
| [github.com/dvsekhvalnov/jose2go](http://github.com/dvsekhvalnov/jose2go) | < 1.6.0 | latest | jsoe2go vulnerable to denial of service via large p2c value | 
| NPM wrangler pkg | >= 3.0.0, < 3.19.0 | 3.19.0 | arbitrary remote code execution within wrangler dev workers sandbox (CRITICAL!) | 
| [golang.org/x/crypto](http://golang.org/x/crypto) | < v0.17.0 | latest | Prefix Truncation Attack against ChaCha20-Poly1305 and Encrypt-then-MAC aka Terrapin |

Also update test go modules as those should be kept current with regular code depenencies.

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #_[issue number]_

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Comment by sicoyle** [comment]
Going to let this sit until closer to code freeze ~May 28th, as this should be part of the release process.

**Review by berndverst — APPROVED** [review]

**Comment by berndverst** [comment]
> Going to let this sit until closer to code freeze ~May 28th, as this should be part of the release process.

If you don't want something merged you should make it a draft PR @sicoyle

**Review by berndverst — CHANGES_REQUESTED** [review]
When you imported `dapr/dapr` 1.13.2 into certification tests it seems that a method signature has changed in Dapr runtime (happens often). You need to update all cert test with the method changes.

After some examination it seems the errors are related to tracing changes. Please resolve this to get this PR merged. Thanks!

Error is this
```# github.com/dapr/dapr/pkg/runtime/hotreload/loader/disk
Error: /home/runner/go/pkg/mod/github.com/dapr/dapr@v1.13.2/pkg/runtime/hotreload/loader/disk/resource.go:33:21: not enough type arguments for type Batcher: have 1, want 2
Error: /home/runner/go/pkg/mod/github.com/dapr/dapr@v1.13.2/pkg/runtime/hotreload/loader/disk/resource.go:48:31: cannot infer T (/home/runner/go/pkg/mod/github.com/dapr/kit@v0.13.1-0.20240402103809-0c7cfce53d9e/events/batcher/batcher.go:50:24)```

**Comment by berndverst** [comment]
Is this still being worked on?

**Comment by sicoyle** [comment]
```
time="2024-06-19T21:58:47.296723824Z" level=info msg="Waiting for all outstanding components to be processed…" instance=fv-az2022-311 scope=dapr.runtime type=log ver=unknown
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x78 pc=0x1b250b9]

goroutine 85 [running]:
github.com/dapr/dapr/pkg/runtime/processor.(*Processor).Init(_, {_, _}, {{{0xc0004b7e46, 0x9}, {0xc0004b7e50, 0x10}}, {{0xc0004b7e70, 0x10}, {0x0, ...}, ...}, ...})
	/home/runner/go/pkg/mod/github.com/dapr/dapr@v1.13.0-rc.1.0.20240606205627-cd2df9098d10/pkg/runtime/processor/components.go:43 +0x159
github.com/dapr/dapr/pkg/runtime/processor.(*Processor).processComponentAndDependents(_, {_, _}, {{{0xc0004b7e46, 0x9}, {0xc0004b7e50, 0x10}}, {{0xc0004b7e70, 0x10}, {0x0, ...}, ...}, ...})
	/home/runner/go/pkg/mod/github.com/dapr/dapr@v1.13.0-rc.1.0.20240606205627-cd2df9098d10/pkg/runtime/processor/components.go:146 +0x4a6
github.com/dapr/dapr/pkg/runtime/processor.(*Processor).processComponents.func1({{{0xc0004b7e46, 0x9}, {0xc0004b7e50, 0x10}}, {{0xc0004b7e70, 0x10}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	/home/runner/go/pkg/mod/github.com/dapr/dapr@v1.13.0-rc.1.0.20240606205627-cd2df9098d10/pkg/runtime/processor/components.go:96 +0x5b
github.com/dapr/dapr/pkg/runtime/processor.(*Processor).processComponents(0xc0000ee2d0, {0x25e0470?, 0xc0003805a0?})
	/home/runner/go/pkg/mod/github.com/dapr/dapr@v1.13.0-rc.1.0.20240606205627-cd2df9098d10/pkg/runtime/processor/components.go:109 +0x128
github.com/dapr/kit/concurrency.(*RunnerManager).Run.func1(0xc0003da180?)
	/home/runner/go/pkg/mod/github.com/dapr/kit@v0.13.1-0.20240523225705-106329e5839f/concurrency/runner.go:78 +0x4b
created by github.com/dapr/kit/concurrency.(*RunnerManager).Run in goroutine 31
	/home/runner/go/pkg/mod/github.com/dapr/kit@v0.13.1-0.20240523225705-106329e5839f/concurrency/runner.go:68 +0x177
exit status 2
```

**Comment by sicoyle** [comment]
@berndverst I believe this is ready. I see one failure, and I think its flaky. Please confirm and lmk :)

**Comment by sicoyle** [comment]
ready ready please 😀

**Review by mikeee — COMMENTED** [review]
https://github.com/dapr/components-contrib/actions/runs/9648369463/job/26609411837#step:22:57

**Comment by berndverst** [comment]
`The source code of the Cloudflare Worker has changed, but the Worker has not been recompiled. Please re-compile the Worker by running 'npm ci && npm run build' in 'common/component/cloudflare/worker-src'`

I think that's the remaining thing that needs to be done - recompile it in that folder and commit the changes. Hopefully the test will then pass.

**Review by berndverst — CHANGES_REQUESTED** [review]
Almost there, just need to recompile the cloudflare worker it seems

```
The source code of the Cloudflare Worker has changed, but the Worker has not been recompiled. Please re-compile the Worker by running 'npm ci && npm run build' in 'common/component/cloudflare/worker-src'
```

**Comment by artursouza** [comment]
There are 3 conf tests failing. I reran if they were flaky but seem unrelated to your changes IMO. Maybe something contrib maintainers can fix in a separate PR.

**Review by artursouza — COMMENTED** [review]
LGTM

File: go.mod [hunk 1]
```diff
@@ -137,10 +137,18 @@ require (
...
-	cloud.google.com/go v0.110.8 // indirect
-	cloud.google.com/go/compute v1.23.1 // indirect
+	github.com/felixge/httpsnoop v1.0.4 // indirect
+	go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc v0.47.0 // indirect
+	go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp v0.47.0 // indirect
+	go.opentelemetry.io/otel/exporters/otlp/otlptrace v1.19.0 // indirect
+	go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc v1.19.0 // indirect
+)
+
+require (
```
Comment (ItalyPaleAle) [inline go.mod:147]: Please make sure there are only 2 `require` blocks: one for direct dependencies and one for indirect. It helps keeping the file tidy

**Review by ItalyPaleAle — CHANGES_REQUESTED** [review]

File: go.mod (hunk 1 above)
Comment (sicoyle) [reply go.mod:147]: I'll try the make mod tidy all, but this is not something I manually added

Comment (ItalyPaleAle) [reply go.mod:147]: Sometimes this happens automatically. Requires manual fixing: merge all blocks into 2, and then do `go mod tidy`

Comment (sicoyle) [reply go.mod:142]: okay I manually consolidated the root go mod and the cert tests one, and then did modtidy-all. My latest commit is the result of this.

**Comment by sicoyle** [comment]
ready @ItalyPaleAle I consolidated the required blocks and then `modtidy-all` so this is the final result from there. Pls lmk if you want further tweaks where applicable.

**Review by ItalyPaleAle — APPROVED** [review]

**Comment by yaron2** [comment]
Snyk checks are failing. @ItalyPaleAle do you have visibility into that? I don't

**Comment by ItalyPaleAle** [comment]
> Snyk checks are failing. @ItalyPaleAle do you have visibility into that? I don't

No, I don't have access to the Dapr organization on Skyk either. I see the same failures in the kit PR that Sam opened  (https://github.com/dapr/kit/pull/96) and I don't have visibility into that either

**Comment by sicoyle** [comment]
ready for merge pls @ItalyPaleAle @berndverst

**Pull request merged**